### PR TITLE
Fix CNS Program iptables for delegated IPAM

### PR DESCRIPTION
**Pull request by nddq**
^

### CODE REVIEW

**Review by rbtr — CHANGES_REQUESTED** [review]
the conflist needs to exist https://github.com/Azure/azure-container-networking/blob/3ecb7fb15a31888a5522743d5ceaccd967e63b59/dropgz/build/cni.Dockerfile#L18

**Comment by nddq** [comment]
> the conflist needs to exist
> 
> https://github.com/Azure/azure-container-networking/blob/3ecb7fb15a31888a5522743d5ceaccd967e63b59/dropgz/build/cni.Dockerfile#L18

For the cns image?

**Comment by rbtr** [comment]
for the dropgz build. you've deleted that unrelated file in this change, it needs to be put back.

File: cns/restserver/internalapi.go [hunk 1]
```diff
@@ -383,45 +384,72 @@ func (service *HTTPRestService) programSNATIPTableRules(req *cns.CreateNetworkCo
...
+	}
 
-	// Check if iptables rules exist already
-	if iptables.RuleExists(iptables.V4, iptables.Nat, iptables.Postrouting, "", iptables.Swift) &&
-		iptables.RuleExists(iptables.V4, iptables.Nat, iptables.Swift, azureDNSUDPMatch, snatPrimaryIPJump) &&
-		iptables.RuleExists(iptables.V4, iptables.Nat, iptables.Swift, azureDNSTCPMatch, snatPrimaryIPJump) &&
-		iptables.RuleExists(iptables.V4, iptables.Nat, iptables.Swift, azureIMDSMatch, snatHostIPJump) {
+	// Only check for existence of other iptables rule if SWIFT chain exists.
+	if chainExist {
+		postroutingToSwiftJumpexist, err := ipt.Exists(iptables.Nat, iptables.Postrouting, "-j", "SWIFT-POSTROUTING")
```
Comment (rbtr) [inline cns/restserver/internalapi.go:401]: why the change from `iptables.Swift` to `"SWIFT-POSTROUTING"`? that really should be a const since it is reused so much

Comment (nddq) [reply cns/restserver/internalapi.go:401]: @tamilmani1989 thinks that we should term it like that instead of SWIFT. I can make it a const.

File: cns/restserver/internalapi.go [hunk 2]
```diff
@@ -383,45 +386,72 @@ func (service *HTTPRestService) programSNATIPTableRules(req *cns.CreateNetworkCo
...
+		}
+	}
+
+	logger.Printf("[Azure CNS] Creating SWIFT Chain ...")
+	err = ipt.NewChain(iptables.Nat, SwiftChainName)
+	if err != nil {
+		return types.FailedToRunIPTableCmd, "[Azure CNS] failed to create SWIFT chain : " + err.Error()
+	}
+	logger.Printf("[Azure CNS] Append SWIFT Chain to POSTROUTING ...")
+	err = ipt.Append(iptables.Nat, iptables.Postrouting, "-j", SwiftChainName)
```
Comment (tamilmani1989) [inline cns/restserver/internalapi.go:435]: what if this rule exist in this chain? will it return failure if rule exists?

**Review by tamilmani1989 — DISMISSED** [review]

**Review by rbtr — APPROVED** [review]
lgtm, aks-e tests are broken and won't pass

**Review by tamilmani1989 — APPROVED** [review]
lgtm

**Comment by rbtr** [comment]
bypassing [broken aks-e check](https://msazure.visualstudio.com/One/_build/results?buildId=58996736&view=logs&j=c19228d3-6ffb-5888-78d8-06d8a51317f2&t=1731d15c-83ec-5657-5358-d889aeb235c4&l=52)
